torch_geometric/nn/aggr/multi.py

```diff
@@ -27,21 +32,59 @@ def __init__(self, aggrs: List[Union[Aggregation, str]],
                              f"'{self.__class__.__name__}' "
                              f"(got '{len(aggrs_kwargs)}', "
                              f"expected '{len(aggrs)}'). Ensure that both "
-                             f"'aggrs' and 'aggrs_kwargs' are consistent")
+                             f"'aggrs' and 'aggrs_kwargs' are consistent.")
 
         self.aggrs = torch.nn.ModuleList([
             aggregation_resolver(aggr, **aggr_kwargs)
             for aggr, aggr_kwargs in zip(aggrs, aggrs_kwargs)
         ])
 
+        self.combine_mode = combine_mode
+        if combine_mode == 'proj':
+            if in_channels is None:
+                raise ValueError(
+                    f"'Combine mode '{combine_mode}' must "
+                    f"have `in_channels` specified (got '{in_channels}').")
+            if out_channels is None:
+                out_channels = in_channels
+            self.lin = Linear(in_channels * len(aggrs), out_channels,
+                              bias=True)
+        else:
+            if (in_channels or out_channels) is not None:
+                raise ValueError("Channel projection is only supported in "
+                                 "the `'proj'` combine mode.")
+
+    def reset_parameters(self):
+        for aggr in self.aggrs:
+            aggr.reset_parameters()
+        if self.combine_mode == 'proj':
+            self.lin.reset_parameters()
+
     def forward(self, x: Tensor, index: Optional[Tensor] = None,
                 ptr: Optional[Tensor] = None, dim_size: Optional[int] = None,
                 dim: int = -2) -> Tensor:
         outs = []
         for aggr in self.aggrs:
             outs.append(aggr(x, index, ptr, dim_size, dim))
-        return torch.cat(outs, dim=-1) if len(outs) > 1 else outs[0]
+
+        return self.combine(outs) if len(outs) > 1 else outs[0]
+
+    def combine(self, inputs: List[Tensor]) -> Tensor:
+        if self.combine_mode in ['cat', 'proj']:
+            out = torch.cat(inputs, dim=-1)
+            return self.lin(out) if hasattr(self, 'lin') else out
+        elif self.combine_mode in ['sum', 'add', 'mul', 'mean', 'min', 'max']:
+            return scatter(torch.stack(inputs, dim=0),
```

Comment:
```suggestion
            return self.reduce(torch.stack(inputs, dim=0),
```
Can we use this?

Reply:
Yes, definitely better!

Reply:
Used `Aggregation` for this. So we have more flexibility. WDYT?